ovsdb/ovsdb-tool.c

```diff
@@ -1497,6 +1497,31 @@ do_check_cluster(struct ovs_cmdl_context *ctx)
         }
     }
 
+    /* Check for db consistency:
+     * The serverid must be in the servers list
+     */
+
+    for (struct server *s = c.servers; s < &c.servers[c.n_servers]; s++) {
+        struct shash *servers_obj = json_object(s->snap->servers);
+        const struct shash_node **nodes = shash_sort(servers_obj);
```

Comment:
We're leaking memory allocated for `nodes`. Also we don't really care about the order so we don't need `shash_sort`. We could just use:
```
struct shash_node *node;
SHASH_FOR_EACH (node, servers_obj) {
...
}
```